_posts/2019-10-02-#16401.md

```diff
@@ -8,7 +8,76 @@ host: jonatack
 
 ## Notes
 
+Today's PR is a Draft WIP implementation of the "package relay" concept first
+described in December 2018 in [issue
+#14895](https://github.com/bitcoin/bitcoin/issues/14895) "Package relay design
+questions" which discusses motivation and design.
+
+Package relay was proposed by [Suhas Daftuar](https://github.com/sdaftuar),
+long-time Bitcoin Core contributor and co-founder of [Chaincode
+Labs](https://chaincode.com), who [has been
+working](https://github.com/bitcoin/bitcoin/pulls?q=is%3Apr+author%3Asdaftuar)
+to improve the privacy and resilience of Bitcoin's peer-to-peer network.
+
+Suhas described today's PR as a proof-of-concept motivation for refactoring
+AcceptToMemoryPoolWorker (ATMP), which he did in [PR
+#16400](https://github.com/bitcoin/bitcoin/pull/16400) "Rewrite
+AcceptToMemoryPoolWorker() using smaller parts" that was merged two weeks ago
+and which lays a foundation for multiple transactions for package relay.
+
+Today's PR was opened at the same time as the ATMP refactoring, thus before it
+was merged, and requires no additional p2p protocol changes.
+
+Suhas' motivation for it begins: *Accepting a single transaction to the mempool
+only succeeds if (among other things) the feerate of the transaction is greater
+than both the min relay fee and the mempool min fee. Consequently, a transaction
+below the minimum fee may not be accepted to the mempool, even if we later learn
+of a transaction with a high fee that depends on it.*
+
+- You can find the minimum relay fee and mempool minimum fee in the codebase by
+git grepping for CheckFeeRate, relayMinFee, DEFAULT_MIN_RELAY_TX_FEE,
```

Comment:
suggestion: put backticks around the code terms to style them as code.

Reply:
done